cpp/dolfinx/fem/DirichletBC.h

```diff
@@ -254,16 +327,46 @@ class DirichletBC
   /// @param[in] scale The scaling value to apply
   void set(xtl::span<T> x, double scale = 1.0) const
   {
-    assert(_g);
-    xtl::span<const T> g = _g->x()->array();
-    for (std::size_t i = 0; i < _dofs0.size(); ++i)
-    {
-      if (_dofs0[i] < (std::int32_t)x.size())
-      {
-        assert(_dofs1_g[i] < (std::int32_t)g.size());
-        x[_dofs0[i]] = scale * g[_dofs1_g[i]];
-      }
-    }
+    std::visit(
```

Comment:
There are much simpler ways to do this, see https://www.cppstories.com/2018/06/variant/.